snapcraft.yaml

```diff
@@ -1626,10 +1637,29 @@ parts:
       ln -s gettext-0.19.8 usr/share/gettext-current
       ln -s gdk-pixbuf-2.0/2.10.0 usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/gdk-pixbuf-current
 
-      # Fix dangling symlink by overwriting it
-      ln -sf lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/libc_malloc_debug.so.0 usr/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/libc_malloc_debug.so
-      # CHECK THIS When changing the core base, this line must be update
-      cp /snap/$CRAFT_EXT_CORE_LEVEL/current/lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/libc_malloc_debug.so.0 lib/$CRAFT_ARCH_TRIPLET_BUILD_FOR/
+      # Check dangling libraries symlinks
+      invalid_links=()
+      core_root_prefix=/snap/$CRAFT_EXT_CORE_LEVEL/current
+      for i in "$CRAFT_PRIME"/{,usr}/lib/{"$CRAFT_ARCH_TRIPLET_BUILD_FOR",}/*.so; do
```

Comment:
Wouldn't we be relying on luck that neither the SDK nor the core snaps provide so files in deeper levels not reached by these globs? For that, `find "$CRAFT_PRIME"/{usr/,}lib -xtype l -name '*.so'` would be more robust.

Reply:
If that's the case, then I recommend a python script. It can be more clever... (like I'm doing with https://github.com/sergio-costas/snap-build-tools.git , remove_common.py)

Reply:
Yeah, I guess remove-common has to be adjusted at this point not to remove the duplicates where a *.so file points to them in the sdk, right?

Reply:
Not necessarily. `remove-common` must be executed before any part has been built, but after all the `stage-packages` have been installed. This way, only duplicated files installed from .deb packages will be removed, but not an apparent duplicated file that was built in any part (thus we can build a new version of a library already being distributed in CoreXX or gnome-XX-YYYY-sdk). The *.so files must be removed in that case too, because those have been installed in STAGE by the stage-package list, so they aren't needed to build any of the packages. The ones that are needed to build are those installed in `build-packages`, which aren't installed in STAGE but in the building container itself.